wiki/DEFINITION_OF_DONE_CRITERIA.md

```diff
@@ -0,0 +1,100 @@
+- It is recommended to open an initial PR with the API design if there are going to be significant differences with reference APIs, or lack of a reference API, to have a discussion about it. 
+- Python class is pickleable
+- APIs use `input_to_dev_array` to accept flexible inputs and check their datatypes
+
+#### Testing 
+
+- Pytests for wrapper functionality against Scikit-learn using relevant datasets
+- Stress tests against reasonable inputs (e.g short-wide, tall-narrow, different numerical precision)
+- Pytests for pickle capability
+- Pytests to evaluate correctness against Scikit-learn on a variety of datasets
+- Add algorithm to benchmarks packages
+
+#### Documentation
+
+- Complete and comprehensive Pydoc strings explaining public API, restrictions, and gotchas
+- Docstrings include references to the underlying algorithm (e.g paper or Scikit-learn algorithm being implemented or a description of the algorithm used if nonstandard).
+
+
+## Algorithm Status
+
+Sometimes it is possible that only some of these requirements are satisfied before an algorithm gets released. We assign a status of `bronze`, `silver`, or `gold` to denote an algorithm's level of completeness and robustness.
+
+Note that a change to an algorithm's classification can go up or down, depending on circumstances. For example, an algorithm that has been considered `Gold status` might drop to a `Silver status` when being updated to a new low-level feature or library. 
+
+### Bronze: 
+- Known design flaws affecting sustainability and/or performance require a significant effort / redesign / refactor
+- C++ checklist / API functionality is complete but Python API has not been built
+- Pure Python API exists without using C++ for processing when a C++ implementation would provide a significant performance boost
+- There are known issues that affect memory / accuracy / stability on some inputs
+- Coverage of automated tests in C++ or Python is minimal and could be significantly improved
+- Outdated / legacy code that requires significant work and refactoring
+
+
+### Silver:
+- Known design flaws affecting sustainability and/or performance are fairly minor.
+- Automated tests in C++ or Python are not as extensive as they should be regarding different datasets / inputs, but they do test general functionality
+- There are known (or potential) bugs that affect memory / accuracy, but only on a small subset of specific inputs
+- There are known optimizations that would provide a significant performance boost over the current implementation
+
+
+### Gold: 
+- Design is robust, maintainable, and performant without any known flaws.
+- Algorithm is as performant as possible without any significant known or potential optimizations over the current implementation.
+- Algorithm is consistently correct and bug free on all datasets.
+- There are no known memory usage limitations resulting from the design choices that could be reasonably refactored.
+
```

Comment:
I would add a difficult requirement that algorithm has been used in other pipelines outside of cuml tests - customers or large scale demos and has been around for a bit of time. Happy to chat more to strategize phrasing, but I wouldn't want to declare something gold until it has been used by non-core team members even if we feel good about it